analyzers/tests/SonarAnalyzer.UnitTest/Metrics/CSharpExecutableLinesMetricTest.cs

```diff
@@ -867,10 +872,188 @@ public static bool Foo(int a, int b)
     public static int Bar() => 42;
 }", 7, 8);
 
+#if NET
+
+        [TestMethod]
+        public void Razor_No_Executable_Lines() =>
+            AssertLineNumbersOfExecutableLinesRazor("""
+<p>Hello world!</p>
+@* Noncompliant *@
+@code
+{
+    private void SomeMethod()
+    {
+    }
+}
+""");
+
+        [TestMethod]
+        public void Razor_FieldReference() =>
+            AssertLineNumbersOfExecutableLinesRazor("""
+@page "/razor"
+@using TestCases
+
+<p>Current count: @currentCount</p> <!-- +1 -->
+
+@currentCount <!-- +1 -->
+
+@code {
+    private int currentCount = 0;
+}
+""", 4, 6);
+
+        [TestMethod]
+        public void Razor_MethodReferenceAndCall() =>
+            AssertLineNumbersOfExecutableLinesRazor("""
+<button @onclick="IncrementCount">Increment</button> <!-- Not counted -->
+<p> @(ShowAmount()) </p> <!-- +1 -->
+
+@code {
+    [Parameter]
+    public int IncrementAmount { get; set; } = 1;
+
+    private void IncrementCount()
+    {
+        IncrementAmount += 1; // +1
+    }
+
+    private string ShowAmount() =>
+        $"Amount: {IncrementAmount}"; // Not counted
+}
+""", 2, 10);
+
+        [TestMethod]
+        public void Razor_PropertyReference() =>
+            AssertLineNumbersOfExecutableLinesRazor("""
+@IncrementAmount <!-- +1 -->
+
+@code {
+    [Parameter]
+    public int IncrementAmount { get; set; } = 1;
+}
+""", 1);
+
+        [TestMethod]
+        public void Razor_HtmlAndCode() =>
+            AssertLineNumbersOfExecutableLinesRazor("""
+<button @onclick="AddTodo">Add todo</button>
+
+<ul>
+    @foreach (var todo in todos) <!-- +1 -->
+    {
+        <li>@todo.Title</li> <!-- +1 -->
+    }
+</ul>
+
+<h3>Todo (@todos.Count(todo => !todo.IsDone))</h3> <!-- +1 -->
+
+@code {
+    private List<ToDo> todos = new();
+
+    private void AddTodo()
+    {
+        var x = LocalMethod(); // +1
+        var y = new DateTime();
+
+        int LocalMethod() => 42;
```

Comment:
There are so many cases we don't count. I've added a question on the team channel to see if anybody remembers about specifications or any documented guidelines on how to compute this.

Reply:
https://docs.sonarsource.com/sonarqube/latest/extension-guide/developing-a-plugin/executable-lines/